src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1454,6 +1454,35 @@ private static IReadOnlyList<DocumentConflict> GetConflictsFor(DocumentsOperatio
             };
         }
 
+        private ChangeVectorEntry[] MergeConflictChangeVectorIfNeededAndDeleteConflicts(ChangeVectorEntry[] documentChangeVector,DocumentsOperationContext context, string key)
+        {
+            ChangeVectorEntry[] mergedChangeVectorEntries = null;
+            bool firstTime = true;
+            int numOfConflicts = 0;
+            foreach (var conflict in GetConflictsFor(context, key))
+            {
+                numOfConflicts++;
+                if (firstTime)
+                {
+                    mergedChangeVectorEntries = conflict.ChangeVector;
+                    firstTime = false;
+                    continue;
+                }
+                mergedChangeVectorEntries = ReplicationUtils.MergeVectors(mergedChangeVectorEntries, conflict.ChangeVector);
+            }
+            //We had conflicts need to delete them
+            if (mergedChangeVectorEntries != null)
+            {
+                DeleteConflictsFor(context, key);
+                Interlocked.Add(ref _hasConflicts,-1*numOfConflicts);
```

Comment:
Hm... yes, that would be better. See the AfterCommitWhenNewReadTransactionsPrevented event.